### DISCUSSION TITLE
How can I Upload my 60 MB files in Github?

### DISCUSSION THREAD

**Discussion by Mehedi-Hasan-soft-web-support** [community/community#141874] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

How can I Upload my 60 MB files in Github?

**Comment by goodLikes** [comment]
**Uploading large files to GitHub: Using Git Large File Storage (LFS)**

GitHub imposes limits on the size of files that can be directly uploaded to the repository. For large files such as video, audio, or large images, it is recommended to use **Git Large File Storage (LFS)**. This tool is specifically designed for storing and managing large files in Git repositories.

**How it works:**

1. **Installing Git LFS:**
      * Install Git LFS following the official instructions: [https://git-lfs.github.com/](https://www.google.com/url?sa=E&source=gmail&q=https://git-lfs.github.com/)
2. **Initialization in the repository:**
* Go to your repository and run the command:
        ```bash
        git lfs install
        ```
3. **Adding files to LFS Tracking:**
* Determine which files you want to track using LFS. For example:
        ```bash
        git lfs track "*.mp4" "*.avi"
        ```
      * This will create a `.gitattributes` file, which will specify the file types tracked by LFS.
4. **Add and commit changes:**
* Add and commit changes as usual:
``bash
        git add .
        git commit -m "Added large files"
        ``
5. **Transferring files to LFS:**
* Run the command to transfer files to LFS:
        ```bash
        git lfs push
        ```

**Important points:**

  * **File Size:** LFS allows you to store files of almost any size.
  * **Speed:** Uploading and downloading large files may take longer than for regular files.
  * **Storage:** The actual files are stored on GitHub LFS servers, and only links to these files are stored in the repository.
  * **Cost:** For private repositories, there are restrictions on the free use of LFS. For large projects, payment may be required.

**Additional recommendations:**

  * **Use `.gitignore`:** To exclude large files that do not need to be tracked, use the `.gitignore` file.
  * **Check the documentation:** Check out the official Git LFS documentation for more information and additional features.

**Example:**

```bash
# Installing Git LFS
git lfs install

# Initialization in the repository
git lfs install

# Tracking .mp4 files
git lfs track "*.mp4"

# Adding a file to the repository
git add my_video.mp4

# Commit changes
git commit -m "Added a video"

# Transferring a file to LFS
git lfs push
```

**Important:** Before starting to work with Git LFS, it is recommended to create a backup copy of your data.

**Additional resources:**

  * **Git LFS Documentation:** [https://git-lfs.github.com/](https://www.google.com/url?sa=E&source=gmail&q=https://git-lfs.github.com/)
  * **GitHub documentation on large files:** <https://docs.github.com/ru/repositories/working-with-files/managing-large-files>

**If you have any additional questions, do not hesitate to ask\!**

**Search keywords:**

  * Git Large File Storage
* GitHub large files
* how to upload a large file to GitHub
* Git LFS instructions

I hope this information will help you successfully upload your files to GitHub\!

**Reply by ebndev** [reply]
Hi @goodLikes, thanks for participating in GitHub Discussions!

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  

You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Please feel encouraged to continue engaging in Discussions as your authentic self though. We appreciate you being a part of the Community 🌟

**Comment by MuxN4** [comment, marked as answer]
@Mehedi-Hasan-soft-web-support If it's build artifacts or generated files thats making the file size huge consider adding them to your .gitignore but if that's not the case install git's Large File Storage depending on your os example if its ubuntu use `sudo apt install git-lfs`
 then initialize it in your repo with `git lfs install`  then track your large files (specify the file pattern)
`git lfs track "*.your-file-extension"`

track your  .gitattributes
git add .gitattributes

Add and commit your files normally
git add `file-name`
git commit -m "large file"
git push origin main

**Reply by Mehedi-Hasan-soft-web-support** [reply]
Thanks For your Suggestions

**Comment by fjyeo** [comment]
Hi mate, not sure what your exact circumstances are but to upload a 60 MB file to GitHub, you'll need Git LFS. First, install it (e.g., `sudo apt install git-lfs` on Ubuntu) and run `git lfs install` in your repo. Then, track your large files with `git lfs track "*.your-file-extension"`. After that, you should be able to add, commit, and push your files like usual.

**Reply by Mehedi-Hasan-soft-web-support** [reply]
Thanks For your Suggestions

**Comment by KDHBuddhika** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> How can I Upload my 60 MB files in Github?
You can upload large files like your 60 MB file to GitHub by using [Git Large File Storage (Git LFS)](https://git-lfs.github.com/). Git LFS is designed to handle large files efficiently. Here's a short guide:
steps:
git lfs install
git lfs track "*.yourfiletype"
git add yourfile
git commit -m "Add large file"
git push origin your-branch
